Boards STM32: add SPI support on Arduino standard spi pin whenever possible

ABOSTM opened the pull request:
Boards STM32: add SPI support on Arduino standard spi pin (D10, D11, D12 and D13) whenever possible.

When necessary (when spi_nss pin not available on D10), use cs-gpio for NSS signal.

ABOSTM committed:
boards: arm: stm32h747i_disco: add Arduino SPI support

Add Arduino SPI support for stm32h747i_disco.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32f769i_disco: add Arduino SPI support

Add Arduino SPI support for stm32f769i_disco.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32f723e_disco: add Arduino SPI support

Add Arduino SPI support for stm32f723e_disco.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32f469i_disco: add Arduino SPI support

Add Arduino SPI support for stm32f469i_disco.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l476rg: add Arduino SPI support

Add Arduino SPI support for nucleo_l476rg.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l433rc_p: add Arduino SPI support

Add Arduino SPI support for nucleo_l433rc_p.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_g431rb: fix Arduino SPI support

Fix Arduino SPI support for nucleo_g431rb.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_g474re: fix Arduino SPI support

Fix Arduino SPI support for nucleo_g474re.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: b_l4s5i_iot01a: full Arduino SPI support

Full Arduino SPI support for b_l4s5i_iot01a

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: disco_l475_iot1: full Arduino SPI support

Full Arduino SPI support for disco_l475_iot1

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f030r8: full Arduino SPI support

Full Arduino SPI support for nucleo_f030r8

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f070rb: full Arduino SPI support

Full Arduino SPI support for nucleo_f070rb

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f103rb: full Arduino SPI support

Full Arduino SPI support for nucleo_f103rb

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f207zg: full Arduino SPI support

Full Arduino SPI support for nucleo_f207zg

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f302r8: full Arduino SPI support

Full Arduino SPI support for nucleo_f302r8

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f334r8: full Arduino SPI support

Full Arduino SPI support for nucleo_f334r8

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f401re: full Arduino SPI support

Full Arduino SPI support for nucleo_f401re

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32l496g_disco: full Arduino SPI support

Full Arduino SPI support for stm32l496g_disco

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f412zg: full Arduino SPI support

Full Arduino SPI support for nucleo_f412zg

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f413zh: full Arduino SPI support

Full Arduino SPI support for nucleo_f413zh

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f429zi: full Arduino SPI support

Full Arduino SPI support for nucleo_f429zi

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f746zg: full Arduino SPI support

Full Arduino SPI support for nucleo_f746zg

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f756zg: full Arduino SPI support

Full Arduino SPI support for nucleo_f756zg

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_f767zi: full Arduino SPI support

Full Arduino SPI support for nucleo_f767zi

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_h743zi: full Arduino SPI support

Full Arduino SPI support for nucleo_h743zi

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_h753zi: full Arduino SPI support

Full Arduino SPI support for nucleo_h753zi

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l412rb_p: full Arduino SPI support

Full Arduino SPI support for nucleo_l412rb_p

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l452re: full Arduino SPI support

Full Arduino SPI support for nucleo_l452re

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l496zg: full Arduino SPI support

Full Arduino SPI support for nucleo_l496zg

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: nucleo_l4r5zi: full Arduino SPI support

Full Arduino SPI support for nucleo_l4r5zi

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: arm: stm32f746g_disco: full Arduino SPI support

Full Arduino SPI support for stm32f746g_disco

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM committed:
boards: shield: x_nucleo_idb05a1: add nucleo_l476rg spi overlay

Standard Arduino CLK signal is on D13,
nucleo_l476rg board has been modified accordingly.
Need an overlay to use default x_nucleo_idb05a1 spin D3 (PB3)

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

tagunil commented on boards/arm/nucleo_l433rc_p/nucleo_l433rc_p.dts:
```diff
@@ -82,6 +82,13 @@
 	status = "okay";
 };
 
+
```
Excessive blank line here.

ABOSTM replied:
Thanks. Fixed.

tagunil commented on boards/shields/x_nucleo_idb05a1/boards/nucleo_l476rg.overlay:
```diff
@@ -0,0 +1,10 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
```
Excessive blank line here.

ABOSTM replied:
Thanks. Fixed

ABOSTM committed:
boards: arm: stm32h747i_disco: rework uart split between CM4 and CM7

Move uart/usart pinctrl to common file.
Set status in dedicated m7 or m4 file.
But only one core can control a peripheral instance.
Arbitrary default choice:
usart1 --> m7
uart8 --> m4

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

ABOSTM commented:
The remaining buildkite issue is: 
https://github.com/zephyrproject-rtos/zephyr/issues/36448

ABOSTM commented:
Rebased to benefit from merge of buildkite fix (#36454)
